digital_input_test.go

```diff
@@ -116,9 +116,16 @@ func TestUpdate(t *testing.T) {
 			Expected: true,
 			HasEvent: false,
 		},
+		{
+			Previous: true,
+			Contents: "0\n",
+			Expected: false,
+			HasEvent: true,
+		},
 	}
 	events := make(chan *DigitalInputReader)
 	defer close(events)
+	i := 0
```

Comment:
Instead of this `i`, you could just use the range operator here instead (in the next line):

`for i, testCase := range cases {}`

Reply:
done, here: 243e8b86dd4d4c5817591201ec30fd82c93f9816